frontend/test/pytest/python_interface/transforms/quantum/test_xdsl_parity_synth.py

```diff
@@ -0,0 +1,491 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit test module for the ParitySynth transform"""
+from collections import namedtuple
+from itertools import product
+
+import numpy as np
+import pytest
+from numpy.testing import assert_allclose, assert_equal
+
+pytest.importorskip("xdsl")
```

Comment:
The `xdsl` marker is being used to split testing into separate tests within the CI workflows. This is a good indicator for us to add better dev documentation. I'll make a note for myself to do that.
```suggestion
pytest.mark.xdsl
pytest.importorskip("xdsl")
```